cmd/commands/common.go

```diff
@@ -279,57 +279,44 @@ func getIngressClassFromUserSelect(ingressClassNames []string) (string, error) {
 	return result, nil
 }
 
-func inferProviderFromRepo(opts *git.CloneOptions) {
-	if opts.Provider != "" {
-		return
-	}
-
-	if strings.Contains(opts.Repo, "github.com") {
-		opts.Provider = "github"
-	}
-	if strings.Contains(opts.Repo, "gitlab.com") {
-		opts.Provider = "gitlab"
-	}
-}
-
-func ensureGitToken(cmd *cobra.Command, cloneOpts *git.CloneOptions, verify bool) error {
-	errMessage := "Value stored in environment variable GIT_TOKEN is invalid; enter a valid runtime token"
+func ensureGitToken(cmd *cobra.Command, gitProvider cfgit.Provider, cloneOpts *apgit.CloneOptions) error {
+	ctx := cmd.Context()
+	errMessage := "Value stored in environment variable GIT_TOKEN is invalid; enter a valid runtime token: %w"
 	if cloneOpts.Auth.Password == "" && !store.Get().Silent {
 		err := getGitTokenFromUserInput(cmd)
-		errMessage = "Invalid runtime token; enter a valid token"
+		errMessage = "Invalid runtime token; enter a valid token: %w"
 		if err != nil {
 			return err
 		}
 	}
 
-	if verify {
-		err := cfgit.VerifyToken(cmd.Context(), cloneOpts.Provider, cloneOpts.Auth.Password, cfgit.RuntimeToken)
+	if gitProvider != nil {
+		err := gitProvider.VerifyToken(ctx, cfgit.RuntimeToken, cloneOpts.Auth.Password)
 		if err != nil {
 			// in case when we get invalid value from env variable TOKEN we clean
 			cloneOpts.Auth.Password = ""
-			return fmt.Errorf(errMessage)
+			return fmt.Errorf(errMessage, err)
 		}
-	}
-
-	if cloneOpts.Auth.Password == "" {
+	} else if cloneOpts.Auth.Password == "" {
 		return fmt.Errorf("must provide a git token using --git-token")
 	}
 
 	return nil
 }
 
-func ensureGitPAT(cmd *cobra.Command, opts *RuntimeInstallOptions) error {
+func ensureGitPAT(ctx context.Context, opts *RuntimeInstallOptions) error {
```

Comment:
Can we add a comment on `ensureGitPat` and `ensureGitToken` explaining what each does, because reading it makes me wonder what is the difference and why we need both.